encled

```diff
@@ -197,6 +262,14 @@ def main(argv):
             if result: return(result)
         return(0)
 
+    # search disk device
+    disk_device = find_serial( argv[1] )  # by serial number
+    if not disk_device:
+        disk_device = find_disk( argv[1] )  # by /dev/disk/
+
+    if disk_device: # a hard disk device has been found
+        argv[1] = disk_device # overwrite argv[1] with
```

Comment:
Changing argv value is not a good practice. Can we just switch the code below (where argv[1]) is used to use of disk_device?

Basically:

```
    disk_device = find_serial( argv[1] )  # by serial number
    if not disk_device:
        disk_device = find_disk( argv[1] )  # by /dev/disk/
    else:
        disk_device = argv[1]
```

(and replace all uses of `argv[1]` with disk device).